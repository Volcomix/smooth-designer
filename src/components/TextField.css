.TextField {
  position: relative;
}

.TextField-input {
  position: absolute !important;
}

.TextField-size {
  white-space: pre;
  font-size: 16px;
  min-height: 48px;
  visibility: hidden;
}
