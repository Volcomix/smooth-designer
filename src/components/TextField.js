//@flow
import React from 'react'
import MuiTextField from 'material-ui/TextField'
import './TextField.css'

export type Props = {
  hintText: string,
  value: string,
  onChange: (event: {}, newValue: string) => void,
  inputRef: (input: ?HTMLInputElement) => void,
}

const TextField = ({ hintText, value, onChange, inputRef }: Props) => (
  <div className="TextField">
    <MuiTextField
      className="TextField-input"
      fullWidth={true}
      hintText={hintText}
      value={value}
      onChange={onChange}
      ref={inputRef}
    />
    <div className="TextField-size">{value}</div>
  </div>
)

export default TextField
