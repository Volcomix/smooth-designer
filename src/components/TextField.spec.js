//@flow
import React from 'react'
import { shallow, mount } from 'enzyme'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'
import TextField, { type Props } from './TextField'

const props: Props = {
  hintText: 'A value',
  value: 'TextField value',
  onChange: jest.fn(),
  inputRef: jest.fn(),
}

it('renders without crashing', () => {
  const wrapper = shallow(<TextField {...props} />)
  expect(wrapper).toMatchSnapshot()
})

it('calls onChange when the input value changes', () => {
  const wrapper = shallow(<TextField {...props} />)
  wrapper.find('.TextField-input').simulate('change', {}, 'New value')
  expect(props.onChange).toHaveBeenCalledWith({}, 'New value')
})

it('gets the input ref', () => {
  const wrapper = mount(
    <MuiThemeProvider>
      <TextField {...props} />
    </MuiThemeProvider>,
  )
  expect(props.inputRef).toHaveBeenCalledWith(
    wrapper.find('TextField.TextField-input').instance(),
  )
})
