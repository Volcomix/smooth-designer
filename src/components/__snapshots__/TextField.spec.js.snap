// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="TextField"
>
  <TextField
    className="TextField-input"
    disabled={false}
    floatingLabelFixed={false}
    fullWidth={true}
    hintText="A value"
    multiLine={false}
    onChange={[Function]}
    rows={1}
    type="text"
    underlineShow={true}
    value="TextField value"
  />
  <div
    className="TextField-size"
  >
    TextField value
  </div>
</div>
`;
